File: admin_flutter/lib/features/liquidity/liquidity_screen.dart
/**
 * TigerWallet Admin - Liquidity Feature
 * Complete implementation with dark/light theme support
 */

import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';

class LiquidityScreen extends StatefulWidget {
  const LiquidityScreen({Key? key}) : super(key: key);

  @override
  State<LiquidityScreen> createState() => _LiquidityScreenState();
}

class _LiquidityScreenState extends State<LiquidityScreen> {
  List<dynamic> _pools = [];
  Map<String, dynamic>? _stats;
  bool _loading = true;
  bool _isDark = false;
  bool _showAddModal = false;

  @override
  void initState() {
    super.initState();
    _loadData();
  }

  Future<void> _loadData() async {
    setState(() => _loading = true);
    try {
      final poolsRes = await http.get(Uri.parse('https://api.tigerwallet.com/admin/v1/liquidity/pools'), headers: {'Authorization': 'Bearer ${await _getToken()}'});
      final statsRes = await http.get(Uri.parse('https://api.tigerwallet.com/admin/v1/liquidity/stats'), headers: {'Authorization': 'Bearer ${await _getToken()}'});
      if (poolsRes.statusCode == 200) _pools = json.decode(poolsRes.body)['data'];
      if (statsRes.statusCode == 200) _stats = json.decode(statsRes.body);
    } catch (e) {
      _pools = _getMockPools();
      _stats = _getMockStats();
    } finally {
      setState(() => _loading = false);
    }
  }

  Future<String> _getToken() async => '';

  List<dynamic> _getMockPools() {
    return [
      {'id': '1', 'pair': 'USDT/ETH', 'token_a': 'USDT', 'token_b': 'ETH', 'reserve_a': 5000000.0, 'reserve_b': 2500.0, 'total_supply': 100000.0, 'apr': 15.5, 'volume_24h': 2500000.0, 'fees_24h': 7500.0, 'status': 'active'},
      {'id': '2', 'pair': 'USDT/BTC', 'token_a': 'USDT', 'token_b': 'BTC', 'reserve_a': 10000000.0, 'reserve_b': 200.0, 'total_supply': 250000.0, 'apr': 12.3, 'volume_24h': 5000000.0, 'fees_24h': 15000.0, 'status': 'active'},
    ];
  }

  Map<String, dynamic> _getMockStats() {
    return {'total_pools': 2, 'total_value_locked': 15000000.0, 'volume_24h': 7500000.0, 'fees_24h': 22500.0};
  }

  Future<void> _addLiquidity(String poolId, double amountA, double amountB) async {
    try {
      await http.post(Uri.parse('https://api.tigerwallet.com/admin/v1/liquidity/pools/$poolId/add'), headers: {'Authorization': 'Bearer ${await _getToken()}'}, body: json.encode({'user_id': 1, 'amount_a': amountA, 'amount_b': amountB}));
      _loadData();
    } catch (e) { /* Handle error */ }
    setState(() => _showAddModal = false);
  }

  void _toggleTheme() => setState(() => _isDark = !_isDark);

  @override
  Widget build(BuildContext context) {
    return Theme(data: _isDark ? ThemeData.dark() : ThemeData.light(), child: Scaffold(
      appBar: AppBar(title: const Text('Liquidity Pools'), actions: [IconButton(icon: Icon(_isDark ? Icons.light_mode : Icons.dark_mode), onPressed: _toggleTheme), IconButton(icon: const Icon(Icons.add), onPressed: () => setState(() => _showAddModal = true))]),
      body: Stack(children: [Column(children: [_buildStats(), Expanded(child: _buildPoolsList())]), if (_showAddModal) _buildAddModal()]),
    ));
  }

  Widget _buildStats() {
    if (_stats == null) return const SizedBox.shrink();
    return Container(padding: const EdgeInsets.all(16), child: Row(children: [
      Expanded(child: _buildStatCard('Total Pools', '${_stats!['total_pools']}')),
      Expanded(child: _buildStatCard('TVL', '\$${(_stats!['total_value_locked'] / 1000000).toStringAsFixed(1)}M')),
      Expanded(child: _buildStatCard('24h Volume', '\$${(_stats!['volume_24h'] / 1000000).toStringAsFixed(1)}M')),
      Expanded(child: _buildStatCard('24h Fees', '\$${(_stats!['fees_24h']).toStringAsFixed(0)}')),
    ]));
  }

  Widget _buildStatCard(String label, String value) {
    return Card(child: Padding(padding: const EdgeInsets.all(12), child: Column(children: [Text(value, style: const TextStyle(fontSize: 18, fontWeight: FontWeight.bold)), Text(label, style: TextStyle(fontSize: 11, color: Colors.grey[600]))])));
  }

  Widget _buildPoolsList() {
    if (_loading) return const Center(child: CircularProgressIndicator());
    return ListView.builder(padding: const EdgeInsets.all(16), itemCount: _pools.length, itemBuilder: (context, index) {
      final pool = _pools[index];
      return Card(margin: const EdgeInsets.only(bottom: 12), child: Padding(padding: const EdgeInsets.all(16), child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
        Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [
          Text(pool['pair'], style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 18)),
          Chip(label: Text(pool['status'].toString().toUpperCase()), backgroundColor: pool['status'] == 'active' ? Colors.green : Colors.grey),
        ]),
        const SizedBox(height: 12),
        Row(children: [_buildPoolDetail('Reserve A', '${pool['reserve_a']} ${pool['token_a']}'), _buildPoolDetail('Reserve B', '${pool['reserve_b']} ${pool['token_b']}'), _buildPoolDetail('APR', '${pool['apr']}%')]),
        const SizedBox(height: 8),
        Row(children: [_buildPoolDetail('24h Vol', '\$${(pool['volume_24h'] / 1000).toStringAsFixed(0)}K'), _buildPoolDetail('24h Fees', '\$${pool['fees_24h']}')]),
        const SizedBox(height: 12),
        Row(children: [ElevatedButton(onPressed: () => _addLiquidity(pool['id'], 1000, 1), child: const Text('Add Liquidity')), const SizedBox(width: 8), OutlinedButton(onPressed: () {}, child: const Text('Remove'))]),
      ])));
    });
  }

  Widget _buildPoolDetail(String label, String value) {
    return Expanded(child: Column(children: [Text(label, style: TextStyle(fontSize: 11, color: Colors.grey[600])), Text(value, style: const TextStyle(fontWeight: FontWeight.w600))]));
  }

  Widget _buildAddModal() {
    return Container(color: Colors.black54, child: Center(child: Card(margin: const EdgeInsets.all(32), child: Padding(padding: const EdgeInsets.all(24), child: Column(mainAxisSize: MainAxisSize.min, children: [
      const Text('Add Liquidity', style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold)),
      const SizedBox(height: 16),
      TextField(decoration: const InputDecoration(labelText: 'Token A Amount'), keyboardType: TextInputType.number),
      const SizedBox(height: 8),
      TextField(decoration: const InputDecoration(labelText: 'Token B Amount'), keyboardType: TextInputType.number),
      const SizedBox(height: 16),
      Row(mainAxisAlignment: MainAxisAlignment.end, children: [TextButton(onPressed: () => setState(() => _showAddModal = false), child: const Text('Cancel')), ElevatedButton(onPressed: () {}, child: const Text('Add'))]),
    ])))));
  }
}

File: admin/android/app/src/main/java/com/tigerwallet/admin/ui/screens/liquidity/LiquidityScreen.kt
package com.tigerwallet.admin.ui.screens.liquidity

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.launch

data class LiquidityPool(val id: Long, val pair: String, val tokenA: String, val tokenB: String, val reserveA: Double, val reserveB: Double, val totalSupply: Double, val apr: Double, val volume24h: Double, val fees24h: Double, val status: String)

class LiquidityViewModel : ViewModel() {
    private val _pools = mutableStateOf<List<LiquidityPool>>(emptyList())
    val pools: List<LiquidityPool> = _pools.value
    private val _stats = mutableStateOf<Map<String, Any>>(emptyMap())
    val stats: Map<String, Any> = _stats.value
    private val _loading = mutableStateOf(false)
    val loading: Boolean = _loading.value
    private val _isDark = mutableStateOf(false)
    val isDark: Boolean = _isDark.value
    private val _showAddModal = mutableStateOf(false)
    val showAddModal: Boolean = _showAddModal.value

    init { loadData() }
    fun loadData() { viewModelScope.launch { _loading.value = true; try { _pools.value = getMockPools(); _stats.value = mapOf("totalPools" to 2, "totalValueLocked" to 15000000.0, "volume24h" to 7500000.0, "fees24h" to 22500.0) } catch (e: Exception) { } finally { _loading.value = false } } }
    fun toggleTheme() { _isDark.value = !_isDark.value }
    fun showAddModal() { _showAddModal.value = true }
    fun hideAddModal() { _showAddModal.value = false }

    private fun getMockPools(): List<LiquidityPool> = listOf(
        LiquidityPool(1, "USDT/ETH", "USDT", "ETH", 5000000.0, 2500.0, 100000.0, 15.5, 2500000.0, 7500.0, "active"),
        LiquidityPool(2, "USDT/BTC", "USDT", "BTC", 10000000.0, 200.0, 2500000.0, 12.3, 5000000.0, 15000.0, "active")
    )
}

@Composable
fun LiquidityScreen(viewModel: LiquidityViewModel = androidx.lifecycle.viewModel.compose.rememberViewModel { LiquidityViewModel() }) {
    val isDark = viewModel.isDark
    val pools = viewModel.pools
    val stats = viewModel.stats
    val loading = viewModel.loading
    val colors = if (isDark) darkColors() else lightColors()

    MaterialTheme(colors = colors) {
        Scaffold(topBar = { TopAppBar(title = { Text("Liquidity Pools") }, actions = { IconButton(onClick = { viewModel.toggleTheme() }) { Icon(if (isDark) "light_mode" else "dark_mode", "Theme") }; IconButton(onClick = { viewModel.showAddModal() }) { Icon("add", "Add") } }, backgroundColor = colors.surface) }, backgroundColor = colors.background) { p ->
            Column(Modifier.padding(p)) {
                Row(Modifier.padding(16.dp).fillMaxWidth()) {
                    StatCard("Pools", "${stats["totalPools"]}", Modifier.weight(1f))
                    StatCard("TVL", "$${((stats["totalValueLocked"] as Double? ?: 0.0) / 1000000).toInt()}M", Modifier.weight(1f))
                    StatCard("24h Vol", "$${((stats["volume24h"] as Double? ?: 0.0) / 1000000).toInt()}M", Modifier.weight(1f))
                    StatCard("24h Fees", "$${(stats["fees24h"] as Double? ?: 0.0).toInt()}", Modifier.weight(1f))
                }
                if (loading) Box(Modifier.fillMaxSize(), Alignment.Center) { CircularProgressIndicator() }
                else LazyColumn(Modifier.padding(16.dp)) {
                    items(pools) { pool ->
                        Card(Modifier.fillMaxWidth().padding(bottom = 12.dp)) {
                            Column(Modifier.padding(16.dp)) {
                                Row(Modifier.fillMaxWidth(), Arrangement.SpaceBetween) {
                                    Text(pool.pair, style = MaterialTheme.typography.titleLarge)
                                    Surface(color = if (pool.status == "active") android.graphics.Color.GREEN else android.graphics.Color.GRAY, shape = MaterialTheme.shapes.small) { Text(pool.status.uppercase(), Modifier.padding(horizontal = 8.dp, vertical = 4.dp), color = android.graphics.Color.WHITE) }
                                }
                                Spacer(Modifier.height(12.dp))
                                Row(Modifier.fillMaxWidth()) {
                                    Col("Reserve A", "${pool.reserveA.toInt()} ${pool.tokenA}", Modifier.weight(1f))
                                    Col("Reserve B", "${pool.reserveB} ${pool.tokenB}", Modifier.weight(1f))
                                    Col("APR", "${pool.apr}%", Modifier.weight(1f))
                                }
                                Spacer(Modifier.height(8.dp))
                                Row(Modifier.fillMaxWidth()) {
                                    Col("24h Vol", "$${(pool.volume24h / 1000).toInt()}K", Modifier.weight(1f))
                                    Col("24h Fees", "$${pool.fees24h.toInt()}", Modifier.weight(1f))
                                    Col("Supply", "${pool.totalSupply.toInt()}", Modifier.weight(1f))
                                }
                                Spacer(Modifier.height(12.dp))
                                Row { Button(onClick = { }) { Text("Add Liquidity") }; Spacer(Modifier.width(8.dp)); OutlinedButton(onClick = { }) { Text("Remove") } }
                            }
                        }
                    }
                }
            }
        }
    }
}

@Composable private fun StatCard(label: String, value: String, m: Modifier) = Card(m = m.padding(4.dp)) { Column(Modifier.padding(12.dp), Alignment.CenterHorizontally) { Text(value, style = MaterialTheme.typography.headlineSmall); Text(label, style = MaterialTheme.typography.caption) } }
@Composable private fun Col(label: String, value: String, m: Modifier) = Column(m, Alignment.CenterHorizontally) { Text(label, style = MaterialTheme.typography.caption); Text(value, style = MaterialTheme.typography.bodyMedium) }
